### PR TITLE
[FLASK] Expose transaction origin to transaction insight snaps

### LINKED ISSUE #16631: [FLASK] Expose transaction origin to Snaps with permission
We added a permission caveat to expose the transaction origin to Snaps in MetaMask/snaps-monorepo#956. This is yet to be implemented in the extension.

### PR DESCRIPTION
**Pull request by FrederikBolding**
## Explanation

Allows transaction insight snaps to access the origin of a transaction by specifying a permission caveat.

This PR also reworks the permission description system to support each permission resulting in multiple "permission descriptions"

Fixes #16631 

## Manual Testing Steps
1. Create a transaction insight snap with the permission `endowment:transaction-insight: { allowTransactionOrigin: true }`
2. Install the snap
3. Propose a transaction
4. See that the snap can access the transaction origin

### CODE REVIEW

**Comment by FrederikBolding** [comment]
Blocked on making a decision on copy for the permission with caveat.

**Review by hmalik88 — DISMISSED** [review]

File: ui/components/app/connected-accounts-permissions/connected-accounts-permissions.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ const ConnectedAccountsPermissions = ({ permissions }) => {
                 className="connected-accounts-permissions__checkbox"
               />
               <label htmlFor={permissionName}>
-                {getPermissionDescription(t, permissionName).label}
+                {getPermissionDescription(t, permissionName)[0].label}
```
Comment (Mrtenz) [inline ui/components/app/connected-accounts-permissions/connected-accounts-permissions.js:58]: Is this guaranteed to be a single item?

**Review by Mrtenz — APPROVED** [review]

File: ui/components/app/connected-accounts-permissions/connected-accounts-permissions.js (hunk 1 above)
Comment (FrederikBolding) [reply ui/components/app/connected-accounts-permissions/connected-accounts-permissions.js:58]: This only applies to DApps right now which only have single item permissions, so for now yeah.

Comment (Mrtenz) [inline ui/components/app/connected-accounts-permissions/connected-accounts-permissions.js:58]: The length of the result should be asserted, and the function should have a `permissionValue` passed to it.

**Review by Mrtenz — CHANGES_REQUESTED** [review]

File: ui/components/app/connected-accounts-permissions/connected-accounts-permissions.js (hunk 1 above)
Comment (FrederikBolding) [reply ui/components/app/connected-accounts-permissions/connected-accounts-permissions.js]: Should be fixed with latest commit

**Review by Mrtenz — APPROVED** [review]

**Review by hmalik88 — APPROVED** [review]

**Pull request merged**